interval-timer.js

```diff
@@ -0,0 +1,70 @@
+import { Timer } from "./timer.js";
+import { is_function } from "./util/type-check.js";
+
+class IntervalTime extends Timer {
+  completed_intervals = [];
+  last_completed_elapsed_time = 0;
+  constructor(
+    on_complete_interval,
+    on_start,
+    on_end,
+    on_update,
+    update_interval_rate
+  ) {
+    super(on_start, on_end, on_update, update_interval_rate);
+    this.on_complete_interval = on_complete_interval;
+  }
+
+  complete_interval() {
+    const elapsed_time = this.get_elapsed_time();
+
+    if (!elapsed_time) {
```

Comment:
We probably want to check a few things here, elapsed time, is_running or if it's paused should throw an error.